<launch>

  <arg name="unique_id" default="$(env ROBOT_NAME)"/>

  <arg name="scan" default="/$(arg unique_id)/laser/scan_filtered"/>    <!-- in m -->
  <arg name="odom" default="/$(arg unique_id)/rb1/mobile_base_controller/odom" /> <!-- NONE, HARRIS, LINES, SUBPIX -->
  <arg name="cmd_vel" default="/$(arg unique_id)/rb1/mobile_base_controller/cmd_vel"/>
  <arg name="move_base" default="/$(arg unique_id)/move_base/move_base"/>
  <arg name="base_link" default="/$(arg unique_id)/base_link"/>
  <arg name="locate_station" default="/$(arg unique_id)/locate_station"/>

  <!-- A simple launch file for the neo_locate_station package. -->
  <group ns="$(arg unique_id)">
    <!-- params -->
    <param name="unique_id" value="/$(arg unique_id)"/>
    <param name="visualize" value="true"/>
    <param name="x_threshold" value="0.055"/>
    <param name="y_threshold" value="0.055"/>
    <param name="nr_of_tries" value="5"/>
    <param name="min_points" value="4"/>
    <param name="max_points" value="20"/>
    <param name="version" value="2"/>
    <param name="dist_pre_goal" value="0.85"/>
    <param name="dist_goal" value="0.6"/>
    <param name="goal_offset_y" value="-0.25"/>
    <param name="approach_velocity" value="0.05"/>
    <param name="dist_points_min" value="0.05"/>
    <param name="dist_points_max" value="0.3"/>
    <param name="high_min" value="0.06"/>
    <param name="high_max" value="0.09"/>
    <param name="min_nr_found" value="6"/>
    <param name="average_calculation_number" value="40"/>
    <param name="min_intensity" value="228"/>
    <param name="use_angle_adjust" value="true"/>
    <param name="stop_laser_dist_threshold" value="0.6"/>

    <node name="neo_locate_station" pkg="neo_locate_station" type="locate_station_node" output="screen">
      <remap from="/sick_s300/scan_filtered" to="$(arg scan)"/>
      <remap from="/odom" to="$(arg odom)"/>
      <remap from="/cmd_vel" to="$(arg cmd_vel)"/>
      <remap from="move_base" to="$(arg move_base)"/>
      <remap from="/base_link" to="$(arg base_link)"/>
      <remap from="/locate_station" to="$(arg locate_station)"/>
    </node>
  </group>
</launch>
